apis/v1beta1/config.go

```diff
@@ -530,6 +525,38 @@ type Telemetry struct {
 	Resource map[string]*string `json:"resource,omitempty" yaml:"resource,omitempty"`
 }
 
+// ToAnyConfig converts the Telemetry struct to an AnyConfig struct.
+func (t *Telemetry) ToAnyConfig() (*AnyConfig, error) {
+	data, err := json.Marshal(t)
+	if err != nil {
+		return nil, err
+	}
+	var result map[string]interface{}
+	if err := json.Unmarshal(data, &result); err != nil {
+		return nil, err
+	}
+
+	normalizeConfig(result)
+
+	return &AnyConfig{
+		Object: result,
+	}, nil
+}
+
+func (t *Telemetry) AddPrometheusMetricsEndpoint(host string, port int32) {
```

Comment:
IMO this method should be a helper function that returns an otelConfig.MetricReader and then we append it to t.Metrics.Readers rather than putting a method on Telemetry